draft-ietf-dnsop-dns-catalog-zones.md

```diff
@@ -460,53 +458,40 @@ zones), but nameservers MUST NOT process such broken zones as catalog
+If all consumers of the catalog zones involed support the `coo` property, it is RECOMMENDED to perform migration of a member zone by following the procedure described in (#cooproperty).
+Otherwise a migration of member zone from a catalog zone `$OLDCATZ` to a catalog zone `$NEWCATZ` has to be done by: first removing the member zone from `$OLDCATZ`; second adding the member zone to `$NEWCATZ` after having made sure all catalog zone consumers of `$OLDCATZ` that are also consumer of `$NEWCATZ` had the member zone removed from `$OLDCATZ`.
 
-The new owner is advised to increase the serial of the member zone after the ownership change, so that the old owner can detect that the transition is done.
-The old owner then removes the member zone from `old.catalog`.
+If in the process of an migration some consumers of the involved catalog zones did not catch the removal of the member zone from `$OLDCATZ` (because of a lost packet or down time or otherwise) they may consider the update with the addition of the member zone in `$NEWCATZ` to be a name clash (see #nameclash) and as a consequence the member is not migrated to `$NEWCATZ`.
+This possibility needs to be anticipated with a member zone migration.
+Recovery from such a situation is out of the scope of this document.
+It may for example entail a manually forced retransfer of `$NEWCATZ` to consumers after they have been detected to have received and processed the removal of the member zone from `$OLDCATZ`.
 
 ## Zone associated state reset {#zonereset}
 
-[FIXME: we now have two mechanisms to reset zones and we should bring that back to one at some point.]
-
-When the `<unique-N>` label of a member zone changes and the zone does not have an
-epoch property (see (#epochproperty)), all its associated state MUST be reset,
-including the zone itself.
-This can be relevant for example when zone ownership is changed.
-In that case one does not want the new owner to inherit the metadata.
-Other situations might be resetting DNSSEC state, or forcing a new zone transfer.
-This reset is tied to `<unique-N>` because A simple removal followed by an addition of the member zone would be insufficient for this purpose because it is infeasible for secondaries to track, due to missed notifies or being offline during the removal/addition.
-
-If an epoch property is present, the steps in (#epochproperty) describe how
-to signal a zone reset.
+It may be desirable to reset state (such as zone data and DNSSEC keys) associated with a member zone.
+If all consumers of the catalog zone support the `epoch` property, it is RECOMMENDED to perform a zone state reset following the procedure described in (#epochproperty).
+Otherwise a zone state reset has to be done by: first removing the member zone from the catalog; add the member zone to the catalog again after having made sure all catalog zone consumers did process the member zone removal.
```

Comment:
similar, doesn't this lead to the zone to be deprovisioned? (may be acceptable in case of reset -- but not sure if it is always acceptable)

Reply:
Yes, you should use `epoch` to overcome that :)